File: Client/Captions/CaptionsLib/source/utils/FileUtils.vb
'Filename: FileUtils.vb
'Version: 20121016

Namespace ClipFlair.CaptionsLib.Utils

  Public NotInheritable Class FileUtils

#Region "Methods"

    Public Overloads Shared Function CheckExtension(ByVal filename As String, ByVal extension As String) As String
      If (filename IsNot Nothing) AndAlso filename.EndsWith(extension, StringComparison.InvariantCultureIgnoreCase) Then Return extension Else Return Nothing
    End Function

    Public Overloads Shared Function CheckExtension(ByVal filename As String, ByVal extensions As String()) As String
      If filename IsNot Nothing Then
        For Each s As String In extensions
          If filename.EndsWith(s, StringComparison.InvariantCultureIgnoreCase) Then Return s
        Next s
      End If
      Return Nothing
    End Function

#End Region

  End Class

End Namespace

File: Client/Captions/CaptionsLib/source/utils/FileUtils.cs
//Filename: FileUtils.cs
//Version: 20131113

using System;

namespace ClipFlair.CaptionsLib.Utils
{

	public static class FileUtils
	{

		#region --- Methods ---

		public static string CheckExtension(string filename, string extension)
		{
			if ((filename != null) && filename.EndsWith(extension, StringComparison.InvariantCultureIgnoreCase))
				return extension;
			else
				return null;
		}

		public static string CheckExtension(string filename, string[] extensions)
		{
			if (filename != null) {
				foreach (string s in extensions) {
					if (filename.EndsWith(s, StringComparison.InvariantCultureIgnoreCase))
						return s;
				}
			}
			return null;
		}

		#endregion

	}

}